mesa/visualization/components/matplotlib.py

```diff
@@ -46,31 +64,53 @@ def SpaceMatplotlib(
 ):
     """Create a Matplotlib-based space visualization component."""
     update_counter.get()
-    space_fig = Figure()
-    space_ax = space_fig.subplots()
+
     space = getattr(model, "grid", None)
     if space is None:
         space = getattr(model, "space", None)
 
     # https://stackoverflow.com/questions/67524641/convert-multiple-isinstance-checks-to-structural-pattern-matching
     match space:
         case mesa.space._Grid():
-            _draw_grid(space, space_ax, agent_portrayal, propertylayer_portrayal, model)
+            fig, ax = draw_orthogonal_grid(
+                space, agent_portrayal, propertylayer_portrayal, model
+            )
+        case OrthogonalMooreGrid():
+            fig, ax = draw_orthogonal_grid(
+                space, agent_portrayal, propertylayer_portrayal, model
+            )
+        case OrthogonalVonNeumannGrid():
+            fig, ax = draw_orthogonal_grid(
+                space, agent_portrayal, propertylayer_portrayal, model
```

Comment:
Can't these all be grouped?  `case OrthogonalGrid():`, `case HexGrid():`, etc?

Reply:
I don't know. I want to do some testing here to make it a bit cleaner.

Reply:
So you can use `|` as an `or` operator, which cleans up the code considerably:

```python
    match space:
        case mesa.space._Grid() | OrthogonalMooreGrid() | OrthogonalVonNeumannGrid():
            draw_orthogonal_grid(space, agent_portrayal, ax)
        case HexSingleGrid() | HexMultiGrid() | mesa.experimental.cell_space.HexGrid():
            draw_hex_grid(space, agent_portrayal, ax)
        case mesa.space.NetworkGrid() | mesa.experimental.cell_space.Network():
            draw_network(space, agent_portrayal, ax)
        case mesa.space.ContinuousSpace():
            draw_continuous_space(space, agent_portrayal, ax)
        case VoronoiGrid():
            draw_voroinoi_grid(space, agent_portrayal, ax
```